From f9430c65f2a20dba1bbcbca1e0470ca6312c86a7 Mon Sep 17 00:00:00 2001
Message-Id: <f9430c65f2a20dba1bbcbca1e0470ca6312c86a7.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Uma Shankar <uma.shankar@intel.com>
Date: Mon, 21 Oct 2013 16:46:10 +0530
Subject: [PATCH 045/312] REVERTME [VPG]: drm/i915: Handled wrong crtc in
 rotation ioctl

Wrong crtc id was not properly handled in rotation
ioctl. Handling the error condition for such cases.

Issue: APDEV-1238
Change-Id: I9b4539e5ce1db86901b4dcc0f0cab9ae2fd79a1d
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
Reviewed-on: http://android.intel.com:8080/139670
Reviewed-by: Stimson, Dale B <dale.b.stimson@intel.com>
Tested-by: Sun, Hang L <hang.l.sun@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Reviewed-on: https://android.intel.com/162825
Reviewed-by: Kulkarni, Vandita <vandita.kulkarni@intel.com>
Tested-by: Kulkarni, Vandita <vandita.kulkarni@intel.com>
Tested-by: sys_abtbuild <sys_abtbuild@intel.com>
Reviewed-by: Bloomfield, Jon <jon.bloomfield@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index d2179b2..cef0fb1 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2499,6 +2499,11 @@ int i915_set_plane_180_rotation(struct drm_device *dev, void *data,
 
 	crtc = obj_to_crtc(obj);
 	DRM_DEBUG_DRIVER("[CRTC:%d]\n", crtc->base.id);
+	if (!crtc->enabled) {
+		DRM_ERROR("[CRTC:%d] not active\n", crtc->base.id);
+		return -EINVAL;
+	}
+
 	intel_crtc = to_intel_crtc(crtc);
 	pipe = intel_crtc->pipe;
 
-- 
1.7.9.5

